src/tqec/templates/atomic/square.py

```diff
@@ -171,3 +172,17 @@ def scale_to(self, k: int) -> "AlternatingCornerSquareTemplate":
     @property
     def shape(self) -> Shape2D:
         return Shape2D(self._dimension.value, self._dimension.value)
+
+    def get_midline_plaquettes(
+        self, horizontal: TemplateOrientation = TemplateOrientation.HORIZONTAL
+    ) -> list[tuple[int, int]]:
+        midline_shape, iteration_shape = self.shape.x, self.shape.y
+        if midline_shape % 2 == 1:
+            raise TQECException(
+                "Midline is not defined for odd "
+                + f"{'height' if horizontal == TemplateOrientation.HORIZONTAL else 'width'}."
+            )
+        midline = midline_shape // 2 - 1
+        if horizontal == TemplateOrientation.VERTICAL:
+            return [(midline, column) for column in range(iteration_shape)]
+        return [(row, midline) for row in range(iteration_shape)]
```

Comment:
Isn't this one already implemented by the `AlternatingRectangleTemplate.get_midline_plaquettes`? Why aren't you swapping `midline_shape` and `iteration_shape` depending on the orientation here?

Reply:
since this is a square `midline_shape == iteration_shape`, and this class inherits from `Template` so we can't inherit the function.

Reply:
Right, I missed that, thanks for the answer :)